projects/dotcms-field-elements/src/components/dot-textarea/dot-textarea.tsx

```diff
@@ -0,0 +1,149 @@
+import { Component, Prop, State, Event, EventEmitter, Method } from '@stencil/core';
+import Fragment from 'stencil-fragment';
+import { DotFieldStatus } from '../../models/dot-field-status.model';
+
+/**
+ * Represent a dotcms textarea control.
+ *
+ * @export
+ * @class DotTextareaComponent
+ */
+@Component({
+    tag: 'dot-textarea',
+    styleUrl: 'dot-textarea.scss'
+})
+export class DotTextareaComponent {
+    @Prop({ mutable: true }) value: string;
+    @Prop() name: string;
+    @Prop() regexcheck: string;
+    @Prop() regexcheckmessage: string;
+    @Prop() label: string;
+    @Prop() hint: string;
+    @Prop() required: boolean;
+    @Prop() requiredmessage: string;
+    @Prop() disabled = false;
+
+    @Event() valueChange: EventEmitter;
+    @Event() statusChange: EventEmitter;
+
+    @State() _valid = true;
+    @State() _dotTouched = false;
+    _dotPristine = true;
+
+    /**
+     * Reset properties of the filed, clear value and emit events.
+     *
+     * @memberof DotTextareaComponent
+     */
+    @Method()
+    reset(): void {
+        this._dotPristine = true;
+        this._dotTouched = false;
+        this.value = '';
+        this._valid = true;
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    componentWillLoad(): void {
+        this.emitStatusChange();
+    }
+
+    hostData() {
+        return {
+            class: {
+                'dot-valid': this.isValid(),
+                'dot-invalid': !this.isValid(),
+                'dot-pristine': this._dotPristine,
+                'dot-dirty': !this._dotPristine,
+                'dot-touched': this._dotTouched,
+                'dot-untouched': !this._dotTouched
+            }
+        };
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
```

Comment:
Why don't we create methods for the attr like:

```
<textarea
    class={this.getClassName(this._valid)}
    name={this.name}
    value={this.value}
    required={this.getRequiredAttr(this.required)}
    onInput={(event: Event) => this.setValue(event)}
    onBlur={() => this.blurHandler()}
    disabled={this.getDisanledAttr(this.disabled)}
/>
```

And that way we can avoid `tslint:disable-next-line:cyclomatic-complexity`